### PR TITLE
update lint rules and fix reported lints

### PR DESCRIPTION
**Pull request by robbecker-wf**
# Overview
There are new lint rules available. Update analysis_options.yaml to use new ones and address any new lints that are reported.

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Raven
<details>
<summary>Number of Findings: 0</summary>

</details>

## Magpie
<details>
<summary>Number of findings: 0</summary>

</details>

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,5 @@
...
-  - sleep 3 # give xvfb some time to start
-script:
-  - pub get
-  - pub run dart_dev format --check
-  - pub run dart_dev analyze
-  - pub publish --dry-run
-  - pub run dart_dev test -p chrome
\ No newline at end of file
+dart_task:
+  - test
```
Comment (robbecker-wf) [inline .travis.yml]: Use the built in dart tasks to run tests! That sure makes it easy.
Just let smithy handle format check, analyze, and pub publish dry run.

Comment (tomconnell-wf) [reply .travis.yml]: It looks like smithy is running tests, too.  Is that desired?  Also, do you want to have travis run analyzer and format?  https://docs.travis-ci.com/user/languages/dart/#Other-Tasks  I guess the pros/cons of each are not clear to me.

**Comment by tomconnell-wf** [comment]
# +1
lint-removal

**Comment by tomconnell-wf** [comment]
# +10
CI passes, and `pub serve test` examples still work.

**Comment by tomconnell-wf** [comment]
# Quality Review Approval: QA +1  
- [x] Testing Instructions
- [x] Dev +1's
- [x] Dev/QA +10 detailing what was tested
- [ ] Tests created, updated, or explained if not present
- [x] All CI checks pass (Smithy, Shipyard, Skynet)
@Workiva/release-management-pp - Rosie, will you automerge?

File: .travis.yml (hunk 1 above)
Comment (robbecker-wf) [reply .travis.yml]: travis is failing with the current setup .. this change at least gets it to pass.
I suppose I could add the other checks in too 🤷‍♂️

**Pull request merged**